src/nvim/path.c

```diff
@@ -2009,10 +2013,10 @@ int path_full_dir_name(char *directory, char *buffer, int len)
 
 // Append to_append to path with a slash in between.
 // Append to_append to path with a slash in between.
-int append_path(char *path, const char *to_append, int max_len)
+int append_path(char *path, const char *to_append, size_t max_len)
 {
-  int current_length = STRLEN(path);
-  int to_append_length = STRLEN(to_append);
+  size_t current_length = STRLEN(path);
```

Comment:
`STRLEN` casts its argument to `char *`, which is unneeded here since `path` is already a `char *`. Since we've refactored away a decent amount of char_u usage in favor of plain char, many cases of `STRLEN` are now unneeded.